spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/interpreter/stages/TypeParser.scala

```diff
@@ -407,6 +408,34 @@ object TypeParser {
                                         isArrayElement: Boolean)
                                        (implicit defaults: Defaults) extends ScalarParser[String]
 
+  private final case class BinaryParser(field: TypedStructField,
+                                        path: String,
+                                        column: Column,
+                                        origType: DataType,
+                                        failOnInputNotPerSchema: Boolean,
+                                        isArrayElement: Boolean)
+                                       (implicit defaults: Defaults) extends PrimitiveParser[Array[Byte]] {
+    override protected def assemblePrimitiveCastLogic: Column = {
+      origType match {
+        case BinaryType => column.cast(BinaryType) // binary: just cast
+        case StringType =>
+          if (field.structField.metadata.contains(MetadataKeys.Encoding)) {
+            val encoding = field.structField.metadata.getString(MetadataKeys.Encoding)
```

Comment:
Would make the `encoding` a property of `BinaryTypeStructField`.

Reply:
True, this helps a bit here (when dealing with the default). Sadly, in the case of the BinaryParser, I still have to pick the encoding from the metadata so the field cannot easily be reused, because I don't decode binary->binary with it. 

Done.